### ISSUE TITLE
implementing expiration logic into extractRehydrationInfo doesn't work 

### ISSUE THREAD

**Issue by dvijeniii05** [reduxjs/redux-toolkit#3661]
Hey all! 

I am using RTKQuery in React-Native app and I am trying to persist querie's cached data.  
Following docs, the persistence from redux-persist + RTKQuery does work.

However, currently I am trying to implement an expiration logic to the persisted data and check if the last query was triggered  more than a certain time ago (let's say 7 days as on the example snippet below), then the data shouldn't be rehydrated and the new query will be triggered once a component is mounted.

I've seen that RTK docs advice on using "extractRehydrationInfo" option to implement a rehydration logic and my 'rough' implementation looks like this:

```ts
  extractRehydrationInfo(action, { reducerPath }) {
    if (action.type === REHYDRATE) {
      const currentTime = Date.now();
      const queryObj = new Object(action.payload.queries);
      const lastQueryTime = Object.entries(queryObj)[0][1].fulfilledTimeStamp;
      const sevenDays = 604800000;
      const isWithinExpiry = lastQueryTime > currentTime - sevenDays;
      if (isWithinExpiry) {
        return action.payload[reducerPath];
      }
      return undefined;
    }
  },
```

However, what is notice is that the rehydration is taken place at all times and the logic of when to 'return action.payload[reducerPath]' is completely ignored. 

So basically the code below also rehydrates the persisted state:
```ts
  extractRehydrationInfo(action, { reducerPath }) {
    if (action.type === REHYDRATE) {
      return undefined;
    }
  },
```

I am a bit confused here, as I assumed that it shouldn't be the case and when we return 'undefined' the rehdyrated data should be set to undefined i.e. not rehydrated.

Am I getting something completely wrong or there is an issue with this option?

Any help or advice is much appreciated !
